domain/model/service/service.go

```diff
@@ -542,3 +589,59 @@ func (s *Service) UpdateCredential(
 
 	return s.st.UpdateCredential(ctx, uuid, key)
 }
+
+// getWatchActivatedModelsMapper returns a mapper function that filters change events to
+// include only those associated with activated models.
+func getWatchActivatedModelsMapper(st State) func(ctx context.Context, db database.TxnRunner,
+	changes []changestream.ChangeEvent) ([]changestream.ChangeEvent, error) {
+
+	return func(ctx context.Context, db database.TxnRunner,
+		changes []changestream.ChangeEvent) ([]changestream.ChangeEvent, error) {
+		modelUUIDs := make([]string, len(changes))
+
+		for i, change := range changes {
+			modelUUID := change.Changed()
+			modelUUIDs[i] = modelUUID
+		}
+
+		// Retrieve all activate status of all models with associated uuids
+		activatedModelUUIDs, err := st.GetActivatedModelUUIDs(ctx, modelUUIDs)
+
+		// There should be no errors returned if there are no activated models found.
+		if err != nil {
+			return nil, err
+		}
+
+		activatedModelUUIDToChangeEventMap := make(map[string]struct{})
+		for _, activatedModelUUID := range activatedModelUUIDs {
+			activatedModelUUIDToChangeEventMap[activatedModelUUID.String()] = struct{}{}
+		}
+
+		activatedModelChangeEvents := make([]changestream.ChangeEvent, 0, len(changes))
+
+		// Add all events associated with activated model UUIDs
+		for i, change := range changes {
+			uuid := modelUUIDs[i]
+			if _, exists := activatedModelUUIDToChangeEventMap[uuid]; exists {
```

Comment:
We can just use `change.Change()` here. By using the modelUUIDs type we risk subtle bugs if something else modifies the order slightly. Better to tie ourselves to the order we are currently running against.

Reply:
Sure thing